### PR TITLE
2547 huobi withdrawal storage

### PR DESCRIPTION
**Pull request by cyberliem**

### CODE REVIEW

File: accounting/huobi/storage/withdrawal-history/postgres/postgres.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+	CREATE TABLE IF NOT EXISTS %[1]s
+(
+	id bigint NOT NULL,
+	time TIMESTAMP NOT NULL,
+	currency TEXT,
+	data JSONB,
+	CONSTRAINT %[1]s_pk PRIMARY KEY(id)
+) ;
+`
+	var (
```
Comment (halink0803) [inline accounting/huobi/storage/withdrawal-history/postgres/postgres.go]: I think this var should be on top of the function before the const.

File: lib/huobi/types.go [hunk 2]
```diff
@@ -51,11 +51,11 @@ type WithdrawHistory struct {
...
 	Amount            float64 `json:"amount"`
 	State             string  `json:"state"`
-	Fees              string  `json:"fees"`
+	Fees              float64 `json:"fees"`
 	ErrorCode         string  `json:"error-code"`
 	ErrorMsg          string  `json:"error-msg"`
-	ToAddress         string  `json:"to-address"`
-	ToAddrTag         string  `json:"to-addr-tag"`
+	ToAddress         string  `json:"address"`
+	ToAddrTag         string  `json:"address-tag"`
```
Comment (halink0803) [inline lib/huobi/types.go]: The field name should match json tag.

File: accounting/huobi/fetcher/fetcher.go [hunk 3]
```diff
@@ -14,14 +15,20 @@ type Fetcher struct {
...
-func NewFetcher(sugar *zap.SugaredLogger, client huobi.Interface, retryDelay time.Duration, attempt int) *Fetcher {
-	return &Fetcher{
+func NewFetcher(sugar *zap.SugaredLogger, client huobi.Interface, retryDelay time.Duration, attempt int, strg ...storage.Interface) *Fetcher {
+	fetcher := &Fetcher{
 		sugar:      sugar,
 		client:     client,
 		retryDelay: retryDelay,
 		attempt:    attempt,
 	}
+	if len(strg) != 0 {
```
[the remark this answers is not in the document]
Comment (cyberliem) [reply accounting/huobi/fetcher/fetcher.go]: no longer used

File: accounting/huobi/storage/withdrawal-history/interface.go [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package withdrawalhistory
+
+import "github.com/KyberNetwork/reserve-stats/lib/huobi"
+
+//Interface is the storage for withdrawal history
+type Interface interface {
+	UpdateWithdrawHistory(withdraw huobi.WithdrawHistory) error
```
[the remark this answers is not in the document]
Comment (cyberliem) [reply accounting/huobi/storage/withdrawal-history/interface.go]: added

File: accounting/huobi/storage/withdrawal-history/postgres/postgres.go (hunk 1 above)
Comment (cyberliem) [reply accounting/huobi/storage/withdrawal-history/postgres/postgres.go]: I found no document specified which is preceded which. Maybe it's our custom rule. Changed nevertheless

File: lib/huobi/types.go (hunk 2 above)
Comment (cyberliem) [reply lib/huobi/types.go]: this is due to the wrong huobi struct definition, which was fixed in a different PR. Currently it's outdated.

File: accounting/huobi/fetcher/fetcher.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (halink0803) [reply accounting/huobi/fetcher/fetcher.go]: I see it is still here. What do you mean "no longer used"?

File: accounting/cmd/accounting-huobi-withdrawal-history/main.go [hunk 5]
```diff
@@ -20,6 +22,7 @@ const (
 	defaultRetryDelay = time.Second
 	fromIDFlag        = "from-id"
 	defaultFromID     = 0
+	defaultPostGresDB = common.DefaultDB
```
Comment (halink0803) [inline accounting/cmd/accounting-huobi-withdrawal-history/main.go]: You can use it directly below, no need to define a constant as a previously constant.

File: accounting/huobi/storage/withdrawal-history/postgres/postgres_test.go [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+	sugar := logger.Sugar()
+
+	db, err := sqlx.Connect("postgres", "host=127.0.0.1 port=5432 user=reserve_stats password=reserve_stats dbname=reserve_stats sslmode=disable")
+	require.NoError(t, err)
+
+	hdb, err := NewDB(sugar, db, WithWithdrawalTableName("test_huobi_withdraw"))
+	require.NoError(t, err)
+
+	defer func() {
+		hdb.TearDown()
```
Comment (halink0803) [inline accounting/huobi/storage/withdrawal-history/postgres/postgres_test.go]: un-handle error.

File: accounting/huobi/storage/withdrawal-history/postgres/postgres_test.go [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+		hdb.Close()
```
Comment (halink0803) [inline accounting/huobi/storage/withdrawal-history/postgres/postgres_test.go]: un-handle error.

File: accounting/cmd/accounting-huobi-withdrawal-history/main.go (hunk 5 above)
Comment (cyberliem) [reply accounting/cmd/accounting-huobi-withdrawal-history/main.go]: the flags const scope is on this main(). Put it on top of the file is to declare the set of default params it's going to run with. Most of our repo use this practice, I'm not sure if we should change it here. Refer to @favadi  for further instruction.

File: accounting/huobi/fetcher/fetcher.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (cyberliem) [reply accounting/huobi/fetcher/fetcher.go]: My bad, it should have been removed. Removed

File: accounting/cmd/accounting-huobi-withdrawal-history/main.go (hunk 5 above)
Comment (favadi) [reply accounting/cmd/accounting-huobi-withdrawal-history/main.go]: I'm OK with either.

**Comment by favadi** [comment]
@cyberliem Can't merge because 96ec1e85033a9d795cdac7480c306cb79536771f is unsigned.

**Pull request merged**